src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ManagedX509ExtensionProcessor.cs

```diff
@@ -128,90 +103,67 @@ public virtual void DecodeX509KeyUsageExtension(byte[] encoded, out X509KeyUsage
                 out bool hasPathLengthConstraint,
                 out int pathLengthConstraint)
         {
-            // BasicConstraints ::= SEQUENCE {
-            //   cA BOOLEAN DEFAULT FALSE,
-            //   pathLenConstraint INTEGER(0..MAX) OPTIONAL }
-
-            DerSequenceReader constraints = new DerSequenceReader(encoded);
-
-            if (!constraints.HasData)
-            {
-                certificateAuthority = false;
-                hasPathLengthConstraint = false;
-                pathLengthConstraint = 0;
-                return;
-            }
-
-            if (constraints.PeekTag() == (byte)DerSequenceReader.DerTag.Boolean)
-            {
-                certificateAuthority = constraints.ReadBoolean();
-            }
-            else
-            {
-                certificateAuthority = false;
-            }
-
-            if (constraints.HasData)
-            {
-                pathLengthConstraint = constraints.ReadInteger();
-                hasPathLengthConstraint = true;
-            }
-            else
-            {
-                pathLengthConstraint = 0;
-                hasPathLengthConstraint = false;
-            }
+            BasicConstraintsAsn constraints = AsnSerializer.Deserialize<BasicConstraintsAsn>(encoded, AsnEncodingRules.BER);
```

Comment:
For all of the places where the deserialization is being done in BER (and it is currently DER in master because it uses DerSequenceReader) please add a test using a BER-but-not-DER value, so that tests don't pass if this is flipped to DER in a later change.

I know we already have one for KeyUsage, but I don't think BasicConstraints or SubjectKeyIdentifier gained one.

Reply:
Yeah, I didn't write the tests yet. Will do it asap.

Reply:
Done